ssas/service/admin/api.go

```diff
@@ -663,6 +663,60 @@ func deleteSystemIP(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
+func createKey(w http.ResponseWriter, r *http.Request) {
+	systemID := chi.URLParam(r, "systemID")
+	trackingID := ssas.RandomHexID()
+	keyEvent := ssas.Event{Op: "CreateKey", TrackingID: trackingID, Help: "calling from admin.createKey()"}
+
+	system, err := ssas.GetSystemByID(systemID)
+	if err != nil {
+		ssas.OperationFailed(keyEvent)
+		jsonError(w, http.StatusNotFound, "Invalid system ID")
+		return
+	}
+
+	var pk ssas.PublicKeyInput
+	if err := json.NewDecoder(r.Body).Decode(&pk); err != nil {
+		ssas.OperationFailed(keyEvent)
+		jsonError(w, http.StatusBadRequest, "Failed to read body")
+		return
+	}
+
+	if err := system.SavePublicKey(strings.NewReader(pk.PublicKey)); err != nil {
+		ssas.OperationFailed(keyEvent)
+		jsonError(w, http.StatusInternalServerError, "Failed to save public key")
+		return
+	}
```

Comment:
How difficult would it be to 1) optionally accept the signed text that DPC currently does 2) if present, validate the signature, rejecting the key if the signature's not valid?

Reply:
@dhgreene probably not too difficult, looking at the Java implementation, it seems pretty straight forward. Will just need to research and find equivalent things to Java